### PR TITLE
Add integration tests for mtcli list-bundles cmd

### PR DESCRIPTION
**Pull request by ashishmax31**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,6 +24,11 @@ all: build
...
 	@go test -count=1 $(PKGS)
 
+
+test-e2e: ## Run e2e integration tests
+	@CGO_ENABLED=1 go build -mod=vendor -a -o $(E2E_MTCLI_PATH) cmd/mtcli/main.go
+	@E2E_MTCLI_PATH=$(E2E_MTCLI_PATH) go test $(INTEGRATION_TESTS)
+
 check: golangci-lint goimports ## Runs all checks.
 	@go fmt $(PKGS)
 	@go vet $(PKGS)
```
Comment (yashvardhan-kukreja) [inline Makefile:34]:
```suggestion
check: golangci-lint goimports ## Runs all checks.
	@go fmt $(PKGS) $(INTEGRATION_TESTS)
	@go vet $(PKGS) $(INTEGRATION_TESTS)
```
Minor nit :)

File: integration/mtcli/list_bundles_test.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	s.Require().NoError(err, "Cant find mtcli binary at E2E_MTCLI_PATH")
+}
+
+func (s *e2eTestSuite) TestMtcliListBundlesE2E() {
+	type testCaseArg struct {
+		indexImageUrl  string
+		expectedOutput []string
+	}
+
+	testCases := []testCaseArg{
```
Comment (yashvardhan-kukreja) [inline integration/mtcli/list_bundles_test.go:45]:
```suggestion
	testCases := []struct{
	        indexImageUrl  string
		expectedOutput []string
	}{
```
Seems like a cleaner place to use anonymous structs from my eyes. WDYT :)

**Comment by yashvardhan-kukreja** [comment]
Good job @ashishmax31 !
Just some tiny aforementioned suggestions, rest all LGTM :)

**Review by yashvardhan-kukreja — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -23,9 +24,14 @@ all: build
 test: ## Run tests.
 	@go test -count=1 $(PKGS)
 
+
+test-e2e: ## Run e2e integration tests
```
Comment (sugarraysam) [inline Makefile:28]: I am a huge fan of testing and integration test, and think it's great work, but I am wondering if this is worth the extra effort of building and running the binary, when you could test the `listBundlesMain(...)` function without any extra toil/preparation? I don't see what we are gaining from running integration tests here vs. regular unit tests of the `listBundlesMain(...)` function

File: integration/mtcli/list_bundles_test.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+				"ocs-osd-deployer.v1.0.0",
+				"ocs-osd-deployer.v1.0.1",
+				"ocs-osd-deployer.v1.0.2",
+				"ocs-osd-deployer.v1.1.0",
+				"ocs-osd-deployer.v1.1.1",
+				"ose-prometheus-operator.v4.8.0",
+			},
+		},
+	}
+	for _, testCase := range testCases {
```
Comment (sugarraysam) [inline integration/mtcli/list_bundles_test.go:67]: Can use `t.Run(...) + t.Parallel(...)` to speedup tests.

File: Makefile (hunk 3 above)
Comment (ashishmax31) [reply Makefile:28]: I think we should have separate unit tests along with integration tests. The idea of this integration test(and future ones) would be to test things from the outside(like how our users would use it) and make sure that the interface we give out to our clients is proper.(Stuff like proper formatting of output, how we take in input through STDIN etc). While I agree that this feature right now is too trivial, the idea still stands.

File: Makefile [hunk 5]
```diff
@@ -23,9 +24,14 @@ all: build
...
+	@CGO_ENABLED=1 go build -mod=vendor -a -o $(E2E_MTCLI_PATH) cmd/mtcli/main.go
+	@E2E_MTCLI_PATH=$(E2E_MTCLI_PATH) go test $(INTEGRATION_TESTS)
+
 check: golangci-lint goimports ## Runs all checks.
-	@go fmt $(PKGS)
-	@go vet $(PKGS)
+	@go fmt $(PKGS) $(INTEGRATION_TESTS)
```
[the remark this answers is not in the document]
Comment (ashishmax31) [reply Makefile:33]: Nope, i separated them out here: https://github.com/mt-sre/addon-metadata-operator/blob/9864ad0bdc74f997de00f3d66775af66aaa8c7f6/Makefile#L12

File: Makefile [hunk 6]
```diff
@@ -23,9 +24,14 @@ all: build
...
+	@go vet $(PKGS) $(INTEGRATION_TESTS)
```
[the remark this answers is not in the document]
Comment (ashishmax31) [reply Makefile:34]: Nope, i separated them out here:
https://github.com/mt-sre/addon-metadata-operator/blob/9864ad0bdc74f997de00f3d66775af66aaa8c7f6/Makefile#L12

File: integration/mtcli/list_bundles_test.go (hunk 4 above)
Comment (ashishmax31) [reply integration/mtcli/list_bundles_test.go:67]: I don't think i can run it in parallel now, the stuff around caching is not thread safe at the moment (file might get overwritten and cause the assertions to fail)

Comment (ashishmax31) [reply integration/mtcli/list_bundles_test.go:67]: testify/suite doesn't support parallel test execution - https://github.com/stretchr/testify/issues/187

File: Makefile (hunk 3 above)
Comment (sugarraysam) [reply Makefile:26]: I am sold :) , thank you

File: Makefile (hunk 5 above)
[the remark this answers is not in the document]
Comment (sugarraysam) [reply Makefile:31]: my mistake !

File: integration/mtcli/list_bundles_test.go (hunk 4 above)
Comment (sugarraysam) [reply integration/mtcli/list_bundles_test.go:67]: interesting, I did not know that

**Pull request merged**